src/GenericFactory.sol

```diff
@@ -85,6 +92,7 @@ contract GenericFactory is IGenericFactory, Owned {
             mstore(lInitCode, add(lByteCodeLength, 0x40))
             mstore(0x40, add(lFreeMem, 0x40))
         }
+        // NB: Releasing back to solidity after the loop is not memory safe.
```

Comment:
Moved from the comment above. Not sure if this is still relevant? We're just returning `lInitCode` after this assembly block so it should be fine? 

I guess it's also possible to do a `return` within the assembly as well?

Reply:
The loop is in solidity with asm in the middle that writes memory without updating the pointer. So the return here is not the problem but the unsound loop is the problem

Reply:
I see.. so since we have fixed the unsound loop [here](url), this comment is no longer relevant, have deleted it

Reply:
can resolve